tests/PHPStan/Analyser/data/bug-6439.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Bug6439;
+
+use function PHPStan\Testing\assertType;
+
+
+class HelloWorld
+{
+	public function unionOnLeft(string $name, ?int $gesperrt = null, ?int $adaid = null):void
+	{
+		$string = 'general';
+		if (null !== $gesperrt) {
+			$string = $string . ' branch-a';
+		}
+		assertType("'general'|'general branch-a'", $string);
+		if (null !== $adaid) {
+			$string = $string . ' branch-b';
+		}
+		assertType("'general'|'general branch-a'|'general branch-a branch-b'|'general branch-b'", $string);
+	}
+
+	public function unionOnRight(string $name, ?int $gesperrt = null, ?int $adaid = null):void
+	{
+		$string = 'general';
+		if (null !== $gesperrt) {
+			$string = 'branch-a ' . $string;
+		}
+		assertType("'branch-a general'|'general'", $string);
+		if (null !== $adaid) {
+			$string = 'branch-b ' . $string;
+		}
+		assertType("'branch-a general'|'branch-b branch-a general'|'branch-b general'|'general'", $string);
```

Comment:
Please test that the limit is not in effect for 31 strings, and is in effect for 32 strings. 32/33 instead is also fine.

Reply:
I added test for the 30/32 elements. I couldn't think of a way to create 33 elements when using just string concat :)